### PR TITLE
ffmpeg-devel: enable av1 support with aom/rav1e

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by i0ntempest** [comment]
Hey, don't mean to rush but have you figured out how should we proceed? In the past few days I've submitted a few new ports that can be used in ffmpeg.

**Comment by mascguy** [comment]
> Hey, don't mean to rush but have you figured out how should we proceed? In the past few days I've submitted a few new ports that can be used in ffmpeg.

Haven't forgotten about you, just preoccupied with some long-overdue tickets/updates. I'll try to revisit this over the next few days though.

**Comment by mascguy** [comment]
Okay, I'm fine with the addition, with one caveat: Given the earlier comment from Ken, re: `rav1e` and `rust`, these additions should only be included for 10.7 and later. (That includes `aom`, as I'm not sure whether that builds on 10.4 and 10.5.)

So if you can make that change, we can move forward.

**Comment by i0ntempest** [comment]
> Okay, I'm fine with the addition, with one caveat: Given the earlier comment from Ken, re: `rav1e` and `rust`, these additions should only be included for 10.7 and later. (That includes `aom`, as I'm not sure whether that builds on 10.4 and 10.5.)
> 
> So if you can make that change, we can move forward.

Alright, so no new variant, just add some logic to link aom/rav1e only on 10.7 and later?
I also want to add SVT-AV1, libvmaf (pure C, one of them uses meson), and libass (added to ffmpeg port before and then removed for unknown reason), what’s your thoughts on these?

**Comment by mascguy** [comment]
> Alright, so no new variant, just add some logic to link aom/rav1e only on 10.7 and later?

Yep!

> I also want to add SVT-AV1, libvmaf (pure C, one of them uses meson), and libass (added to ffmpeg port before and then removed for unknown reason), what’s your thoughts on these?

Let's start with what you have here, before piling anything more on.

File: multimedia/ffmpeg-devel/Portfile [hunk 1]
```diff
@@ -235,6 +235,18 @@ platform darwin {
     if {${os.major} < 12} {
         configure.args-append --disable-indev=avfoundation
     }
+
+    # av1 codecs, available on 10.7+
+    if {${os.major} >= 12} {
```
Comment (mascguy) [inline multimedia/ffmpeg-devel/Portfile:240]: I think you meant 11 here?

**Review by mascguy — CHANGES_REQUESTED** [review]

**Comment by i0ntempest** [comment]
Aight I've made the change. I added SVT-AV1 since it's also an av1 encoder, for now it builds on everything except 10.7, errors in our buildbot log suggests it's AVX2 related. I don't know if this indicates that build bot doesn't support AVX2. I'll see if I can figure that one out, if not I think I'll bump the requirement up by a bit.
libvmaf stuff can be discussed later, libass is already in and I somehow missed it, so ignore that.

**Comment by mascguy** [comment]
> Aight I've made the change. I added SVT-AV1 since it's also an av1 encoder, for now it builds on everything except 10.7, errors in our buildbot log suggests it's AVX2 related. I don't know if this indicates that build bot doesn't support AVX2. I'll see if I can figure that one out, if not I think I'll bump the requirement up by 1.
> libvmaf stuff can be discussed later, libass is already in and I somehow missed it, so ignore that.

Sounds good! And worst-case, we can just enable for 10.8 and later for now, while we sort out the 10.7 issue.

**Comment by i0ntempest** [comment]
> Sounds good! And worst-case, we can just enable for 10.8 and later for now, while we sort out the 10.7 issue.

Good news, blacklisting some old clang versions worked, it just cleared 10.7 buildbot.

**Review by mascguy — APPROVED** [review]

**Pull request merged**